src/GraphQL/GraphQLBuilderExtensions.cs

```diff
@@ -523,6 +559,14 @@ public static IGraphQLBuilder AddClrTypeMappings(this IGraphQLBuilder builder, A
         }
         #endregion
 
+        #region - AddAutoClrMappings -
+        public static IGraphQLBuilder AddAutoClrMappings(this IGraphQLBuilder builder, bool mapInputTypes = true, bool mapOutputTypes = true)
+        {
+            builder.Services.Register<IGraphTypeMapping>(_ => new AutoRegisteringGraphTypeMapping(mapInputTypes, mapOutputTypes), ServiceLifetime.Singleton);
```

Comment:
`AddAutoSchema` uses `TryRegister` and adds `AutoRegisteringGraphTypeMapping` with `true, true` params. Depending on the registration order of `AddAutoClrMappings/ AddAutoSchema` code will work in different ways.

Reply:
This is why I did not remove the `TypeMappings` as you suggested above -- because then this would be an issue.  `AddAutoSchema` needs a way to register a catch-all, but cannot override any manually registered mappings, either by `AddAutoClrMappings` or elsewhere.

How about this - we design the `IGraphTypeMapping` delegate to be somewhat like middleware, where all of the delegates can decide whether or not to change the mapping?  So `AddAutoClrMappings` will only change the mapping if it's currently `null`, whereas mappings added elsewhere would always set the mapping.  Then it should generally not matter which order the mappings were registered in.

```cs
/// <summary>
/// Provides a mapping from CLR types to graph types.
/// </summary>
public interface IGraphTypeMapping
{
    /// <summary>
    /// Returns a graph type for a given CLR type, or <see langword="null"/> if no mapping is available.
    /// Should return <paramref name="preferredGraphType"/> if this instance does not wish to change the mapping.
    /// </summary>
    /// <param name="clrType">The CLR type to be mapped.</param>
    /// <param name="isInputType">Indicates whether the type is an input type.</param>
    /// <param name="preferredGraphType">The graph type that is suggested for this CLR type.</param>
    Type? GetGraphTypeFromClrType(Type clrType, bool isInputType, Type? preferredGraphType);
}
```

Then code would be something like this:

```cs
Type? mappingType = null;
foreach (var mapping in mappings)
{
    mappingType = mapping.GetGraphTypeFromClrType(clrType, isInputType, mappingType);
}

if (mappingType == null && clrType.IsEnum)
{
    mappingType = typeof(EnumerableGraphType<>).MakeGenericType(clrType);
}

//ditto for scalar types

return mappingType;
``

The other thing to consider is that I believe we should retain the ability to add mappings from within the `Schema` as is possible now.  I would suggest this be done by creating a list of `IGraphTypeMapping` implementations within `Schema` and then appending them to those retrieved from DI when passed to `SchemaTypes`.  Is this what you were thinking also?

Reply:
See updated code to reflect this idea.

Reply:
> Then it should generally not matter which order the mappings were registered in.

It will matter if more then one `IGraphTypeMapping` implementations want to change mappings.

Reply:
Keep as is?

Reply:
I would keep as it is now, similar to the above sample, not as is was when the first comments here were written.